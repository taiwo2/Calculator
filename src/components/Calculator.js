import React from 'react';
import App from './App';
import './style.css';

function Calculator() {
  return (
    <div className="calculate__wrapper">
      <h2 className="centre">Lets do some Math</h2>
      <div className="calculator__box">
        <App />
      </div>
    </div>
  );
}

export default Calculator;
