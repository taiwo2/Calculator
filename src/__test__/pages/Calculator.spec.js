import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../../components/Calculator';

test('should correctly render Calculator page', () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
