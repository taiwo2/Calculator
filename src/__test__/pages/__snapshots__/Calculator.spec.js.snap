// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Calculator page 1`] = `
<div
  className="calculate__wrapper"
>
  <h2
    className="centre"
  >
    Lets do some Math
  </h2>
  <div
    className="calculator__box"
  >
    <div>
      <div
        className="display"
        data-testid="show-panel"
      >
        0
        
        
      </div>
      <div
        className="calculator "
        data-testid="button-panel"
      >
        <div
          className="nums-container "
        >
          <div
            className="top-btn"
          >
            <button
              className="light-grey"
              data-testid="button"
              onClick={[Function]}
            >
              AC
            </button>
            <button
              className="light-grey"
              data-testid="button"
              onClick={[Function]}
            >
              +/-
            </button>
            <button
              className="light-grey"
              data-testid="button"
              onClick={[Function]}
            >
              %
            </button>
          </div>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            7
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            8
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            9
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            4
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            5
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            6
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            1
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            2
          </button>
          <button
            className="dark-grey false"
            data-testid="button"
            onClick={[Function]}
          >
            3
          </button>
          <button
            className="dark-grey big-h"
            data-testid="button"
            onClick={[Function]}
          >
            0
          </button>
          <button
            className="light-grey"
            data-testid="button"
            onClick={[Function]}
          >
            .
          </button>
        </div>
        <div
          className="ops-container"
        >
          <button
            className="orange"
            data-testid="button"
            onClick={[Function]}
          >
            /
          </button>
          <button
            className="orange"
            data-testid="button"
            onClick={[Function]}
          >
            x
          </button>
          <button
            className="orange"
            data-testid="button"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="orange"
            data-testid="button"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="orange"
            data-testid="button"
            onClick={[Function]}
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
